grammars/rainfall/ocaml.py

```diff
@@ -0,0 +1,467 @@
+from autoplan.grammar import Rule
+from .labels import GeneralRainfallLabels
+
+Labels = GeneralRainfallLabels
+
+class CleanFirst(Rule):
+    def render(self):
+        recursion = self.choice('recursion', {'' : 1, 'rec' : 1})
+        parameter_specs = self.choice('parameter_specs', {'list_name_only' : 1, 'list_name_and_type' : 1})
+        list_name = self.choice('list_name', {'alon': 1, 'aloi': 1, 'alod' : 1, 'alof' : 1, 'amounts' : 1})
+        _type = self.choice('_type', {'int': 1, 'float': 1})
+        helper_name = self.choice('helper_name', {'rain_helper' : 1, '_rainfall' : 1, 'rains' : 1, 'get_data' : 1})
+        helper_in_body = self.choice('helper_in_body', {True: 1, False: 1})
+        head = self.choice('head', {'head' : 1, 'hd' : 1}) 
+        tail = self.choice('tail', {'tail' : 1, 'tl' : 1, '_' : 1})
+
+        template = '''
+{%- set params -%}
+    {%- if parameter_specs == 'list_name_only' -%}
+        {{list_name}} = 
+    {%- else -%}
+        ({{list_name}} : {{_type}} list) : {{_type}} =
+    {%- endif -%}
+{%- endset -%}
+
+{%- set helper_body -%}
+let {{recursion}} {{helper_name}} ({{list_name}} : {{_type}} list) : {{_type}} list =
```

Comment:
Could add a `uses_annotation` variable for whether the the types are annotated or not (i.e. `(a : b)` vs just `a`).